cli/src/config.rs

```diff
@@ -55,6 +55,7 @@ pub struct Config {
     pub core: CoreConfig,
     pub auth: Option<AuthConfig>,
     pub vault: Option<VaultConfig>,
+    pub release_info: Option<ReleaseInfo>,
```

Comment:
I think we can call it `update_check` or `new_update_check`, so in the config file it will be:
```toml
[update_check]
last_checked_at = "xxx"
```

Reply:
what do you think?

Reply:
im thinking to keep as release_info, reason being is that im storing the release info in here not only update_check. The intend for now is to check for updates for now but we might have a future use case and keeping the naming as release info makes more sense. 

And i've updated the logic; the idea is regardless of github response we take the result from release info and build the logic so if there is a failure on github we still compare with the latest one on config because version always goes up

Reply:
I still don't see why we need to store the values here, if there is a failure on GitHub api, then we shouldn't show the *potentially outdated* info to the user, we should just show nothing. 

What are the future use case that need these values?

Reply:
So my understanding of version number is always are incremental so last cached value is higher then the current version i wouldn't say its really outdated data just that there is a chance that there might be a higher version.

But doesnt mean the upgrade message shows isn't true. So circling back to main intention of showing the upgrade option is that user is using an outdated version

Reply:
And future uses cases i can think of now is, giving options to the user to upgrade within out CLI
As we have all the data we need its just matter of handling the update